eiger_io/fs_handler_dask.py

```diff
@@ -0,0 +1,144 @@
+import dask.array as da
+from pims import FramesSequence
+from databroker.assets.handlers_base import HandlerBase
+import re
+import h5py
+
+import os
+
+
+class PIMSDask(FramesSequence):
+    ''' the dask version of PIMS, takes dask array.
+
+        Notes
+        -----
+            - this should eventually become a PR into PIMS.
+            - this should be upgraded to allow nested FramesSequences
+                (need to allow for defining axes etc)
+    '''
+    # the regexp patterns for expected files
+    # here it is just file containing "master" but could potentially be
+    # expanded upon
+    def __init__(self, data, md=None):
+        '''
+            Initialized a lazy loader for EigerImages
+            Parameters
+            ----------
+            data : dask.array.Array
+                the data
+            md : dict, optional
+                the dictionary of metadata
+        '''
+        self._data = data
+        self._md = md
+
+    @property
+    def md(self):
+        return self._md
+
+    def get_frame(self, i):
```

Comment:
@danielballan need your input. Should it be return Frame(...) here rather than pure numpy array?

Reply:
Long answer: At a scipy conference, a numpy core dev asked, "Who here has subclassed `ndarray`?....Who regrets it?" I regret that `Frame` exists. I think that the most-used feature of `Frame` is its integration with IPython's display hook to display an image as a PNG rather than the standard `ndarray` text `__repr__` but there are other ways to achieve that. Its other feature, the way that it carries `md` along with it, does not survive mathematical operations -- i.e. what happens to `frame.md` when you do `frame` + `other_frame`? What about more complex operations like a rolling average?

So: if this class calls itself a PIMS reader, it should probably return `Frame` for consistency with other PIMS readers. On the other hand, I think it would be fine to experiment with simply returning a plain `ndarray` and see how we like that....